numpy/core/arrayprint.py

```diff
@@ -250,7 +251,7 @@ def _array2string(a, max_line_width, precision, suppress_small, separator=' ',
         data = _leading_trailing(a)
     else:
         summary_insert = ""
-        data = ravel(a)
+        data = ravel(asarray(a))
```

Comment:
There was no problem for matrices.  I only noticed that `ravel(a)` failed for when `a` is of type `<class 'test_multiarray.Rop__radd__'>`.  And `numpy/core/tests/test_multiarray.py:TestBinop` is the only test that fails with `ravel(a)`.  I figured that instead of changing the test, I should change `ravel(a)` to `ravel(asarray(a))` since then the `_array2string` function would have the same exact effect as before this PR (`ravel` used to call `asarray` anyway).